### DISCUSSION TITLE
Any downsides to using supabase's cron?

### DISCUSSION THREAD

**Discussion by codepeach** [supabase/supabase#5604] [Questions] [answered]
I want to run a cron job every 10 minutes to delete rows from my table. Are there any downsides to using supabase's inbuilt cron server as opposed to an external one?

**Comment by GaryAustin1** [comment, marked as answer]
Seems like a perfect use case.
I assume you are talking about pg_cron which is just a Postgres extension.  If you google that you will find a few articles, but none saying avoid it that I have seen.

This one points out a few limitations, that won't apply to what you want to do:
https://fatdba.com/2021/07/30/pg_cron-probably-the-best-way-to-schedule-jobs-within-postgresql-database/

Here is a more technical deep dive:
https://www.citusdata.com/blog/2020/10/31/evolving-pg-cron-together/

**Reply by Anan7Codes** [reply]
Hi, I'm looking to delete rows based on time intervals as well. Basically, I wish to run a cron job every 30 mins and if a row has been existing for more than 5 minutes I wish to delete that row. Could you share an example query that I'd have to run on the SQL editor, to set this up?

**Reply by GaryAustin1** [reply]
Googling "postgres cron job to delete posts after minutes", or something similar,  should find you lots of examples.

**Reply by Anan7Codes** [reply]
Cool thanks :)

**Reply by Anan7Codes** [reply]
Hi Gary, I enabled pg_cron on my dashboard and I ran the query 
"select cron.schedule('* 5 * * * *', '$$ DELETE from public.test_cron where now() - created_at > interval '5 min' $$);" 
on my sql editor but it doesn't seem to work. I'm basically trying to run a cron job every 5 mins and see if my row has been created for over 5 minutes and if yes, delete it.

**Reply by GaryAustin1** [reply]
I see several things.  
You don't have the name of your cron task first ('name','*/5.......
Not sure if the formatter ate part of your code. ` '*/5 * * * *'` is every 5 minutes.
And I don't know if 'min' works versus 'minutes'.

**Reply by Anan7Codes** [reply]
-- select cron.schedule('Delete cached values', '*/5 * * * *', $$ DELETE from public.test_cron where now() - created_at > interval '5 min' $$); --

Ran this query and it works. Thanks ! :)

**Comment by justinekatebe69** [comment]
Does running consistent cron jobs add any cost to my plan on supabase? Lets say I want to run one every 1 minute for an indefinite period. Would I get charged for it?

**Reply by GaryAustin1** [reply]
Not if they don't make external calls to other API's.

**Comment by maxmckenzie** [comment]
One downside i've encountered is when adding in a cron job as a migration to run a scheduled edge function you do something like the below.

```sql
select
  cron.schedule(
    'invoke-function-every-minute',
    '* * * * *', -- every minute
    $$
    select
      net.http_post(
          url:='https://project-ref.supabase.co/functions/v1/function-name',
          headers:='{"Content-Type": "application/json", "Authorization": "Bearer YOUR_ANON_KEY"}'::jsonb,
          body:=concat('{"time": "', now(), '"}')::jsonb
      ) as request_id;
    $$
  );
  ```

The above wants you to hard code your anon key and the URL to call that function. So if i include the above in a migration file. It wont work with my local supabase instance as it points to prod or whatever the hard coded value was, and if we rotate the anon key it'll fail?

**Reply by GaryAustin1** [reply]
Right now the most common solution is to store keys and urls in Vault.  At least then you have one place to go change.  Then where you use them you select them from the Vault view into a variable to use.
